src/cargo/util/config.rs

```diff
@@ -154,6 +154,21 @@ impl Config {
         }
     }
 
+    pub fn get_path(&self, key: &str) -> CargoResult<Option<PathBuf>> {
+        if let Some((string, path)) = try!(self.get_string(&key)) {
+            let string_as_path = Path::new(&string);
+            if string_as_path.is_absolute() || string_as_path.components().count() == 1 {
+                Ok(Some(PathBuf::from(string_as_path)))
+            } else {
+                let prefix_path = try!(path.parent().and_then(|p| p.parent())
+                                       .ok_or(internal("unexpected config file path")));
+                Ok(Some(prefix_path.join(string_as_path)))
+            }
```

Comment:
Too right. I failed to note that http://doc.rust-lang.org/nightly/std/path/struct.PathBuf.html#method.push "just works" with absolute paths. Nice.

I'll revert it back to the previous and much simpler logic.